File: result/repair/transcoder/python2java/java_repair/FIND_MEDIAN_ROW_WISE_SORTED_MATRIX.java
  int f_gold ( int [ ] [ ] m , int r , int c ) {
  int mi = Integer . MAX_VALUE ;
  int mx = - Integer . MAX_VALUE - 1 ;
  for ( int i = 0 ;
  i < r ;
  i ++ ) {
    if ( m [ i ] [ 0 ] < mi ) mi = m [ i ] [ 0 ] ;
    if ( m [ i ] [ c - 1 ] > mx ) mx = m [ i ] [ c - 1 ] ;
  }
  int desired = ( r * c + 1 ) / 2 ;
  while ( ( mi < mx ) && ( mx > mi ) ) {
    int mid = mi + ( mx - mi ) / 2 ;
    int [ ] place = new int[1];
    for ( int i = 0 ;
    i < r ;
    i ++ ) {
      int j = upperBound ( m [ i ] , mid ) ;
      place [ 0 ] = place [ 0 ] + j ;
    }
    if ( place [ 0 ] < desired ) mi = mid + 1 ;
    else mx = mid ;
  }
  System . out . println ( "Median is" + mi ) ;
  return mi ;
}

File: dataset/python2java/python/FIND_MEDIAN_ROW_WISE_SORTED_MATRIX.py
def f_gold ( m , r , c ) :
    mi = sys.maxsize
    mx = - sys.maxsize - 1
    for i in range ( r ) :
        if m [ i ] [ 0 ] < mi :
            mi = m [ i ] [ 0 ]
        if m [ i ] [ c - 1 ] > mx :
            mx = m [ i ] [ c - 1 ]
    desired = ( r * c + 1 ) // 2
    while ( mi < mx ) :
        mid = mi + ( mx - mi ) // 2
        place = [ 0 ]
        for i in range ( r ) :
            j = upper_bound ( m [ i ] , mid )
            place [ 0 ] = place [ 0 ] + j
        if place [ 0 ] < desired :
            mi = mid + 1
        else :
            mx = mid
    print ( "Median is" , mi )
    return mi